src/WebJobs.Extensions.DocumentDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -83,6 +92,13 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
                 {
                     throw new InvalidOperationException("The monitored collection cannot be the same as the collection storing the leases.");
                 }
+
+                if (attribute.CreateIfNotExists)
+                {
+                    // Not disposing this because it might be reused on other Trigger since Triggers could share lease collection
+                    IDocumentDBService service = _config.GetService(leasesConnectionString);
+                    CreateIfNotExistAsync(service, leaseCollectionLocation.DatabaseName, leaseCollectionLocation.CollectionName, attribute.CollectionThroughput).Wait();
```

Comment:
I tried, but this the entire method returns a `Task` I cannot call `await` because it's not `async`.